echo/index.js

```diff
@@ -254,15 +275,29 @@ function Client() {
     debug('request options', requestOptions);
 
     request.get(requestOptions, function onResp(err, response, rawBody) {
-      var statusCode = response && response.statusCode
-        ? response.statusCode
-        : 0;
-
+      var statusCode = response && response.statusCode ? response.statusCode : 0;
       if (err || parseInt(statusCode / 100) !== 2) {
-        error('[echoClient] queryAnalytics error', { err: err, body: rawBody, statusCode: statusCode });
-        callback(err || 'error response status code: ' + statusCode);
+        var errorCode = getErrorCode(err, statusCode);
+  
+        var errorResponse = {
+          code: errorCode.toString(),
+          label: errorCodesAndLabels[errorCode],
+        };
+
+        error('[echoClient] queryAnalytics error', errorResponse);
+        callback(errorResponse);
+        return;
       } else {
-        parseJSON(rawBody, callback);
+        delete rawBody.statusCode;
+        var body;
+        try {
+          body = JSON.parse(rawBody);
```

Comment:
`body` will be `null` is the parsing wasn't successful. There needs to be a check to see if the parsing was successful before returning a response.

Reply:
if you look a couple of lines down, we exit the function and callback with an error, we never return body as part of the response, so this code should be ok